pkg/scaffold/ansible/dockerfile.go

```diff
@@ -39,7 +39,7 @@ func (d *Dockerfile) GetInput() (input.Input, error) {
 	return d.Input, nil
 }
 
-const dockerFileAnsibleTmpl = `FROM quay.io/water-hole/ansible-operator
+const dockerFileAnsibleTmpl = `FROM quay.io/operator-framework/ansible-operator
```

Comment:
In the Dockerfile scaffolds for helm and ansible, should we pin to an image tag? Something like the following?

```go
d.ImageTag = strings.TrimSuffix(version.Version, "+git")
```

If so, we'll have to be careful to make sure our e2e tests use the just-built image vs an existing tagged one.

Reply:
Good question. Our e2e tests will definitely need to build a dev ansible/helm base image from the PR before testing.
For our tagged releases for the SDK we definitely want to use ansible/helm base images with the same image tag as the SDK release version in the scaffold.

But when using the SDK on the master branch, do we want to scaffold base images for the last semver release, or use base images built from the master. I think that we'll need to do the latter.

Right now it seems like we don't have a release tag so it will always use the latest updated tag.

Reply:
I agree, if we're testing a PR on master we should build base images from master. We might want to add a step in the release process for building, tagging, and pushing ansible and helm base images as per @shawn-hurley's previous [comment](https://github.com/operator-framework/operator-sdk/pull/531#discussion_r220566473) and #555.

Reply:
I actually think that the tag should just be the release that is for that operator-sdk binary. I wouldn't add the git because it should be the release number I think right?

Reply:
Here are my thoughts:
* During CI, we can use whatever tag we want with the CI-built image. Ideally we name it something that we never actually push to avoid accidentally using an image we don't intend to. I propose `:dev`. So no matter what branch or tag Travis is building, we'll build `:dev` as our base image locally.
* Next, during the e2e tests, we want to use that `:dev` image. So, after running `operator-sdk new` but before `operator-sdk build`, I would suggest always replacing the scaffolded image name in `build/Dockerfile` with `quay.io/operator-framework/ansible-operator:dev`
* Finally, when we go to push the image tags, we'll run the following for each `$PUSH_TAG`:
    ```sh
    docker tag quay.io/operator-framework/ansible-operator:dev quay.io/operator-framework/ansible-operator:$PUSH_TAG
    docker push quay.io/operator-framework/ansible-operator:$PUSH_TAG
    ```

As for what image tag to scaffold, it sounds like we agree that if `version.Version` does not end with `+git`, then we'll use `version.Version`. The biggest question in my mind is what to scaffold when it does contain `+git`. The options in my mind are:
- Use `:master`.
  - Pro: If users are running operator-sdk from master, they'll get the most recent base image changes.
  - Con: If users are running operator-sdk from an unreleased commit prior to the last stable release, it's possible that the master base image will be incompatible with the rest of the scaffolded operator.
- Just drop `+git`.
  - Pro: No matter what build of operator-sdk is being used, it's likely that the base image will work, since we'll use the "closest" one.
  - Con: If using operator-sdk from master, the latest base image changes will not be present.
- Use a broken placeholder, like `REPLACE_TAG`:
  - In this case, the operator will fail to build, forcing the decision onto the user.

Reply:
Ok, so I am getting confused. I am going to lay out what I think are the significant parts of this that matter.

**Scaffold Changes**
1. The scaffold should only ever use the version (without git).

**CI Changes**
2. The CI build will update the `build/Dockerfile` to use the correct base image (whatever that is) after `operator-sdk new` is called. 
3. Makes sense to use something like `:dev` as the tag for CI when your running the docker build or operator-sdk build. 
4. I agree that we should then tag and push at the end of this process.

Reply:
That all makes sense to me. I think I agree about just always using `strings.TrimSuffix(version.Version, "+git")` for the scaffolded base image tag.

Other opinions?

Reply:
> The scaffold should only ever use the version (without git).

@shawn-hurley I agree on not having an ansible/helm base image with the tag `:v0.3.0+git` but what about `:master`?

Because users who use the SDK binary from the master branch to scaffold a helm/ansible project will get a `build/Dockerfile` that points uses the base image from the last release e.g `quay.io/operator-framework/ansible-operator:v0.3.0`.

I'm fine with doing that but then do we still maintain a `:master` base image tag for users wh want to use the base image with the latest changes from the master, in between releases?

Reply:
I would think that on every PR merge, that we rebuild the images and push to a `:canary` image. That is what we have used in the past. 

Also, did I miss the part of updating latest tag to the newest tag, or is that going to be a manual process?

Reply:
Also, I'd suggest `:master` for a couple of reasons.

- It's more obvious to users where it came from.
- Because it just comes from the branch name, it makes the scripts more flexible. If we wanted to started building another branch (say we have another big integration effort like `controller-runtime`), it would be as easy as updating the `if:` condition in the .travis.yml deploy stage.

Reply:
@joelanford @hasbro17 I personally think that a new ansible or helm operator always uses the latest release version (i.e 0.3.0), not master. I think that if a user wants to use latest or master that is an explicit change to use it in their dockerfile.

Reply:
That sounds fine to me. So long as we maintain a `:master` tag it's not too much work for users to use the `:master` base image when they want to.

@joelanford WDYT? We always keep the `:latest` tag pointing to the latest release. 
And in the scaffold we always use the latest release tag e.g `:v0.3.0`. (Can't use `:latest` I think because that can break older projects when we make a new release image and change latest).

Reply:
Agreed. I'll update the PR with this change.